### PR TITLE
[docs] custom managed workflow

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Initial attempt to reshape our docs to reflect that using Expo is now a continuous process that doesn't ever fork into two different workflows.

## Goals

- Instead of switching contexts between bare/managed, provide the narrative that all projects can use Expo Go, Expo SDK, EAS, and Expo CLI.
- Instead of ejecting, teach prebuilding.
- Stick to teaching about Expo CLI for the entire workflow.

### CODE REVIEW

File: docs/pages/workflow/customizing.md [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
-To eject to the bare workflow, you can run `expo eject` and follow the instructions. Head over to the [bare workflow walkthrough](../bare/exploring-bare-workflow.md) to learn more about what the workflow will look like after ejecting.
+You can do this in a single command:
 
-> 💡 We recommend upgrading to the latest SDK version before ejecting. It will be more difficult to upgrade your app after ejecting because you will also be responsible for native iOS and Android related upgrade steps.
+<TerminalBlock cmd={['# Build your native iOS project', 'expo run:ios', '# Build your native Android project', 'expo run:android']} />
+
+<!-- TODO: Add a doctor command bootstrap easier -->
+<!-- TODO: Don't link outside of Expo docs -->
+
+- `expo run:ios` requires Xcode (Macos only) installed on your computer. See the [setup guide](https://reactnative.dev/docs/environment-setup)
```
Comment (brentvatne) [inline docs/pages/workflow/customizing.md:16]:
```suggestion
- `expo run:ios` requires Xcode (macOS only) installed on your computer. See the [setup guide](https://reactnative.dev/docs/environment-setup)
```

File: docs/pages/workflow/customizing.md [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+
+<!-- TODO: Add a doctor command bootstrap easier -->
+<!-- TODO: Don't link outside of Expo docs -->
+
+- `expo run:android` requires Android Studio setup on your computer. See the [setup guide](https://reactnative.dev/docs/environment-setup)
```
Comment (brentvatne) [inline docs/pages/workflow/customizing.md:21]:
```suggestion
- `expo run:android` requires Android Studio and the Android SDK to be installed. See the [setup guide](https://reactnative.dev/docs/environment-setup)
```

**Review by brentvatne — COMMENTED** [review]
this is off to a really good start!

File: docs/pages/workflow/customizing.md [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
+> Run commands were introduced in SDK 41, prebuilding and running in earlier SDKs may not work as well.
+
+<!-- TODO: Link to run commands doc -->
+<!-- TODO: Link to prebuild commands doc -->
+
+Using the run commands will initially [prebuild](https://expo.fyi/prebuilding) your project to generate all of the native code. If you manually modify the `ios/` or `android/` folders, you won't be able to safely re-run `expo prebuild`, this is known as the [bare workflow](../introduction/managed-vs-bare.md#bare-workflow).
+
+Your app can still run in Expo Go, but any custom native code won't be accessible if it's not already present in the Expo Go app. [Learn more](./using-expo-client).
+
+If you install a package with a [**config plugin**](../guides/config-plugins), you'll need to add the plugin to your `app.json`s [`plugins`](../versions/latest/config/app/#plugins) array, then re-run `expo prebuild` to sync the changes before rebuilding the native app. Often this does things like adding required permissions to the `Info.plist` or `AndroidManifest.xml`.
```
Comment (EvanBacon) [inline docs/pages/workflow/customizing.md:32]:
```suggestion
If you install a package with a Expo [**config plugin**](../guides/config-plugins), you'll need to add the plugin to your `app.json`s [`plugins`](../versions/latest/config/app/#plugins) array, then re-run `expo prebuild` to sync the changes before rebuilding the native app. Often this does things like adding required permissions to the `Info.plist` or `AndroidManifest.xml`.
```

File: docs/pages/workflow/customizing.md [hunk 4]
```diff
@@ -1,11 +1,44 @@
...
+
+If you install a package with a Expo [**config plugin**](../guides/config-plugins), you'll need to add the plugin to your `app.json`s [`plugins`](../versions/latest/config/app/#plugins) array, then re-run `expo prebuild` to sync the changes before rebuilding the native app. Often this does things like adding required permissions to the `Info.plist` or `AndroidManifest.xml`.
+
+You may need to run `expo prebuild --clean` depending on how complex the plugin is. If you've made manual modifications to your `ios/` or `android/` folder, you'll need to manually setup new packages as running `expo prebuild` may not work as expected (think of this like running `yarn` after manually modifying your `node_modules/` folder).
+
+If you want to make static changes to your native project files like the iOS `Info.plist`, or `AndroidManifest.xml` and still have access to prebuilding, check out the [config plugins guide](../guides/config-plugins/#creating-a-plugin)
+
+# Production
+
+When you're ready to ship your app, you can build it with EAS Build!
```
Comment (brentvatne) [inline docs/pages/workflow/customizing.md:40]:
```suggestion
When you're ready to ship your app, you can build it with EAS Build! [Learn more](/build/introduction.md).
```

File: docs/pages/workflow/customizing.md [hunk 5]
```diff
@@ -1,11 +1,44 @@
...
+<!-- TODO: Don't link outside of Expo docs -->
+
+- `expo run:android` requires Android Studio and the Android SDK to be installed. See the [setup guide](https://reactnative.dev/docs/environment-setup)
+
+> Run commands were introduced in SDK 41, prebuilding and running in earlier SDKs may not work as well.
+
+<!-- TODO: Link to run commands doc -->
+<!-- TODO: Link to prebuild commands doc -->
+
+Using the run commands will initially [prebuild](https://expo.fyi/prebuilding) your project to generate all of the native code. If you manually modify the `ios/` or `android/` folders, you won't be able to safely re-run `expo prebuild`, this is known as the [bare workflow](../introduction/managed-vs-bare.md#bare-workflow).
```
Comment (brentvatne) [inline docs/pages/workflow/customizing.md:28]:
```suggestion
Using the run commands will initially [prebuild](https://expo.fyi/prebuilding) your project to generate all of the native code within your project directory. If you manually modify the `ios/` or `android/` folders, you won't be able to safely re-run `expo prebuild`, this is known as the [bare workflow](../introduction/managed-vs-bare.md#bare-workflow).
```

**Review by brentvatne — APPROVED** [review]

File: docs/pages/workflow/customizing.md [hunk 6]
```diff
@@ -1,11 +1,44 @@
...
 
-The process of ejecting is easily reversible so don't worry about experimenting with it. Just commit any changes you want to keep and then follow along with this guide. If you decide you want to abort it, just check out your most recent commit. You can also reverse it in the future by deleting the `ios` and `android` directories.
+The Expo Go app enables us to move quickly by shipping a reusable native runtime for developing apps. If you want to use custom native code that isn't already in the Expo Go app, you will need to build a new native app.
 
-To eject to the bare workflow, you can run `expo eject` and follow the instructions. Head over to the [bare workflow walkthrough](../bare/exploring-bare-workflow.md) to learn more about what the workflow will look like after ejecting.
+You can do this in a single command:
 
-> 💡 We recommend upgrading to the latest SDK version before ejecting. It will be more difficult to upgrade your app after ejecting because you will also be responsible for native iOS and Android related upgrade steps.
+<TerminalBlock cmd={['# Build your native iOS project', 'expo run:ios', '# Build your native Android project', 'expo run:android']} />
```
Comment (brentvatne) [inline docs/pages/workflow/customizing.md:11]:
```suggestion
<TerminalBlock cmd={['# Build and run your native iOS project locally', 'expo run:ios', '# Build and run your native Android project locally', 'expo run:android']} />
```

**Review by brentvatne — APPROVED** [review]

File: docs/pages/bare/hello-world.md [hunk 7]
```diff
@@ -36,39 +38,38 @@ npm install expo-web-browser
...
+  );
 }
 ```
 
 This will not yet work because we haven't linked the native code that powers it. To do this, we need to follow the instructions in the [`expo-web-browser` README](https://github.com/expo/expo/tree/master/packages/expo-web-browser) to configure it for iOS and Android. Let's do it.
 
 ### iOS configuration
 
-Bare projects are initialized using [CocoaPods](https://cocoapods.org/), a dependency manager for iOS projects.
+<TerminalBlock cmd={['# Build your native iOS project', 'expo run:ios']} />
```
Comment (brentvatne) [inline docs/pages/bare/hello-world.md:64]:
```suggestion
<TerminalBlock cmd={['# Build and run your native iOS project', 'expo run:ios']} />
```

**Pull request merged**